### PR TITLE
Adding feature to allow for config files to be written to and read from

### PR DESCRIPTION
**Pull request by allenhsieh**

### CODE REVIEW

File: src/Config/OssIndexServerConfig.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+export class OssIndexServerConfig extends Config {
+  constructor(readonly username: string, readonly token: string) {
+    super(username, token);
+  }
+
+  public saveConfigToFile(
+    saveLocation: string = this.getSaveLocation()
+  ): boolean {
+    let ableToWrite = false;    const stringToSave = `Username: ${this.username}
```
Comment (DarthHater) [inline src/Config/OssIndexServerConfig.ts]: Newline?

Comment (DarthHater) [reply src/Config/OssIndexServerConfig.ts]: Also I'd make a quick lil `getStringToSave(): string` helper method rather than do this.

Comment (allenhsieh) [reply src/Config/OssIndexServerConfig.ts]: @DarthHater oh! not done, just making a draft PR for now!

File: src/Config/OssIndexServerConfig.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    Password: ${this.token}`;
+
+    try {
+      writeFileSync(saveLocation, stringToSave, {flag: 'wx'});
+    } finally { 
+      ableToWrite = true;
```
Comment (DarthHater) [inline src/Config/OssIndexServerConfig.ts]: finally will catch execute no matter what. You probs want to do `ableToWrite = true` after writeFileSync, and then `catch (e) { logError(e) }`

Comment (allenhsieh) [reply src/Config/OssIndexServerConfig.ts]: in: https://github.com/sonatype-nexus-community/auditjs/pull/98/commits/aace2d4a68211f9d1c8d1acb689776cfa49477b5

File: src/Config/OssIndexServerConfig.ts (hunk 1 above)
Comment (allenhsieh) [reply src/Config/OssIndexServerConfig.ts]: in: https://github.com/sonatype-nexus-community/auditjs/pull/98/commits/aace2d4a68211f9d1c8d1acb689776cfa49477b5

**Pull request merged**